### PR TITLE
Various health check improvements + tests

### PR DESCRIPTION
**Pull request by cipherboy**
This adds various small improvements to the PKI health checks, plus a more comprehensive test suite.

Probably best viewed commit by commit.

### CODE REVIEW

File: command/healthcheck/healthcheck.go [hunk 1]
```diff
@@ -110,12 +110,12 @@ func (e *Executor) Execute() (map[string][]*Result, error) {
 		}
 
 		if err := checker.FetchResources(e); err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to execute %v: %w", checker.Name(), err)
```
Comment (stevendpclark) [inline command/healthcheck/healthcheck.go:113]:
```suggestion
			return nil, fmt.Errorf("failed to fetch resources %v: %w", checker.Name(), err)
```

File: command/healthcheck/healthcheck.go [hunk 2]
```diff
@@ -110,12 +110,12 @@ func (e *Executor) Execute() (map[string][]*Result, error) {
...
 		}
 
 		results, err := checker.Evaluate(e)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to execute %v: %w", checker.Name(), err)
```
Comment (stevendpclark) [inline command/healthcheck/healthcheck.go:118]:
```suggestion
			return nil, fmt.Errorf("failed to evaluate %v: %w", checker.Name(), err)
```

**Review by stevendpclark — APPROVED** [review]
👍 some small nits but please feel free to ignore them if you don't agree.

File: command/healthcheck/healthcheck.go (hunk 1 above)
Comment (cipherboy) [reply command/healthcheck/healthcheck.go:113]: I kinda wanted to leave this a little vague for the caller -- they don't really care if it fails in the fetch stage or early in the evaluation stage I think. Both are fundamentally the same step of the health check, to them, its only in implementation that we make that distinction I feel?

Let me know if you disagree :-)

Comment (stevendpclark) [reply command/healthcheck/healthcheck.go:113]: Yup that is fine as well, it was only to differentiate if we get a generic error back from an end-user we could zero in a little more on what step actually had failed.

**Review by stevendpclark — APPROVED** [review]

**Pull request merged**